examples/vue-full-v1/renderer/+config.h.ts

```diff
@@ -6,7 +6,7 @@ export default {
   passToClient: ['pageProps', 'title'],
   clientRouting: true,
   prefetchStaticAssets: 'viewport',
-  onHydrationEnd,
+  onHydrationEnd, // NOTE(aurelien): this assignment breaks if we stop supporting the sync variant
```

Comment:
See inline comment. It seems like we're introducing a breaking change by not supporting the sync variant of void-returning hooks. So I think we shouldn't do that either.

My suggestion is that we keep `OnHydrationEnd | OnHydrationEndSync`, yet from now on, as you said, we try to make examples use async variants as much as we can. What do you think?